src/clients/detector.rs

```diff
@@ -7,12 +7,13 @@ use crate::config::ServiceConfig;
 
 const DETECTOR_ID_HEADER_NAME: &str = "detector-id";
 
-#[cfg_attr(test, derive(Default))]
-#[derive(Clone)]
+#[cfg_attr(test, faux::create)]
+#[derive(Clone, Default)]
```

Comment:
I think adding `Default` here can be risky, since it may assign default values to underlying objects, and we may not be expecting that in certain cases or it may cause weird edge cases. If there are no default then the error would be easier to track. So for this instance, is there a reason to add `Default` here and if that is for testing purposes, can we go with adding that only for test, like [here](https://github.com/foundation-model-stack/fms-guardrails-orchestrator/blob/main/src/config.rs#L46)

Reply:
@gkumbhat I'm not sure why, but when `derive(Default)` is compiled only in the test scope, the following message is shown on VSCode:

```
struct `detector::DetectorClient` has no field named `clients`
field does not existrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2Fhome%2Fmdevino%2Fdev%2Fprojects%2Fwxpe%2Forchestrator%2Fsrc%2Fclients%2Fdetector.rs)
detector.rs(12, 12): `detector::DetectorClient` defined here
```

`cargo check` runs fine, but the following error is shown for `cargo test`:

```
error[E0560]: struct `detector::DetectorClient` has no field named `clients`
  --> src/clients/detector.rs:13:5
   |
12 | pub struct DetectorClient {
   |            -------------- `detector::DetectorClient` defined here
13 |     clients: HashMap<String, HttpClient>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ field does not exist

For more information about this error, try `rustc --explain E0560`.
```

Reply:
@gkumbhat It seems this issue only happens when `derive(Default)` is added before `faux::create`:
In other words, `#[cfg_attr(test, derive(Default), faux::create)]` gives the error I mentioned above, but `#[cfg_attr(test, faux::create, derive(Default))]` works. Changes implemented!